### DISCUSSION TITLE
Welcome to ws-ephemeral Discussions!

### DISCUSSION THREAD

**Discussion by dhruvinsh** [dhruvinsh/ws-ephemeral#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by SpoooodaLs** [comment]
Thank you for this, I'm currently stuck with Windscribe for another year, and didn't realize they only port forward for 7 days. (how annoying)
This will make the rest of that time more bearable!

**Reply by dhruvinsh** [reply]
Yeah I have no complain with Winscribe in term of reliability. But I feel that if they have auto-renewal for port forwarding as a feature that would be a great!

Ephemeral port has advantage regarding the privacy as well, so hey we can't complain much :)

**Comment by chicocheco** [comment]
Hi, thank you for the project! This would not work with 2FA enabled, correct?

**Reply by dhruvinsh** [reply]
Hey, yes as of right now it will not work. But in a week I am planning to implement that feature.

**Reply by chicocheco** [reply]
Hey! You replied superfast! I don't know if it is just a coincidence but maybe I found a temporal workaround. I disabled the 2FA. Run the script (ONESHOT=true) and it stored cookies on my server. I enabled it again and it seems to be still working, I got a different port. It could work probably until the cookies expire? Anyway, thank you. This is great that you are planning on implementing that too.

Also, I don't know if it is necessary but I believe that in order to make it use the new port, you must restart qbittorrent? You do not mention that so I assume it should work without restarting it.

**Reply by dhruvinsh** [reply]
> it stored cookies on my server. I enabled it again and it seems to be still working,

Loophole, nice.

> you must restart qbittorrent?

I have seen other saying they need to. For me always have worked without that.

**Reply by chicocheco** [reply]
Yep, you are right. No need to restart qbit. It starts working on it own after a few minutes. On the other hand, last night, I found out my VPN stopped working, as every new torrent added got stuck at "stalled" until I restarted it. This was immediately resolved with manually rebooting the server but as you can guess, that is not a solution. Did you ran into this kind of issue? I assume that Windscribe dropped an IP address that was being in use and Wireguard did not pick it up. If I read it correctly, Wireguard works this way, that it resolves the IP only once, when you start the interface. I am currently trying to figure out how to re-resolve the IP from time to time. Anyway, I guess that is just another topic that does not belong here.

**Comment by cueballkrill** [comment]
Hi there,

Right, installed (first part is a win!) but now I am absolutely clueless on how to setup qbittorrent to make sure it runs through ws-ephemeral.  I'm on day 2 of unraid, pure windows before this.

Pointers would be greatly appreciated

**Reply by dhruvinsh** [reply]
Nice, enjoy the unraid journey. It will be fun.

Now get ready to read some docs. For qBitorrent, i personally use binhex qBitorrent with vpn. So I would suggest referring to that document. Once you have proper setup you can refer back to ws-ephemeral doc and that will give you a better idea.

**Reply by cueballkrill** [reply]
Thanks for the reply.

I spent a good 4 hours today trying to get to binhex ver with vpn working with windscribe, I'm too dependant on click here do that.   I then went to the stardard version of qbittorrent.  This is synced with ephemeral (changes the port automatically) but I am assuming this is still not vpn'd.

I'm also looking at gluetun, but have managed to crash that!  -_-

**Comment by mleenorris** [comment]
So, I'm feeling a bit dense, but does the container automatically grab a new port every 7 days? or do I have to rerun every 7 days?
